haptics/src/web.ts

```diff
@@ -11,30 +11,38 @@ export class HapticsWeb extends WebPlugin implements HapticsPlugin {
     super({ name: 'Haptics' });
   }
 
-  impact(options: HapticsImpactOptions): void {
+  async impact(options: HapticsImpactOptions): Promise<void> {
     console.log('impact options', options);
+    return Promise.resolve();
   }
 
-  notification(options: HapticsNotificationOptions): void {
+  async notification(options: HapticsNotificationOptions): Promise<void> {
     console.log('notification options', options);
+    return Promise.resolve();
   }
 
-  vibrate(options?: VibrateOptions): void {
+  async vibrate(options?: VibrateOptions): Promise<void> {
     if (navigator.vibrate) {
       const duration = options?.duration || 300;
       navigator.vibrate(duration);
+      return Promise.resolve();
+    } else {
+      return Promise.reject('Vibrate API not available in this browser')
```

Comment:
If you `throw` in an `async` function it rejects automatically

```suggestion
      throw new Error('Vibrate API not available in this browser');
```

Reply:
I know, but as I said, I think it's better (more consistent) to return the promise explicitly for consistency with the native code